### DISCUSSION TITLE
[enhancement] configurable timeout on connect

### DISCUSSION THREAD

**Discussion by Stono** [wevm/wagmi#186] [Ideas]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

0.2.12

### Current Behavior

Currently if my wallet is connected to a network and that network is down (for example; at the time of writing https://matic-mumbai.chainstacklabs.com is down), then the `connect` function does nothing.

It appears to eventually time out, after a good 3-4 minutes:

![Screenshot 2022-02-18 at 16 10 50](https://user-images.githubusercontent.com/1486729/154719601-bff0835d-14b4-4653-8f9b-7d3542f46a77.png)

### Expected Behavior

It'd be nice to have the connect function timeout / reject after a configurable amount of time. Right now it just looks like my site is broken, so I'm keen to feed back to the user that the issue is with the network they've currently got selected in metamask.

### Steps To Reproduce

Set your metamask wallet to a network that's done, any RPC address that times out will do.
Attempt to connect using an injected connector
Wait

**Comment by tmm** [comment]
@Stono converted to discussion as this sounds more like a feature.

**Reply by Stono** [reply]
Totally fair :) Wasn't sure where best to raise it - sorry!

**Reply by 0x9090** [reply]
Would love to see timeouts be configurable too. It would greatly increase the responsiveness of our apps
